### PR TITLE
🐛 retrieve stakeholder info from stakeholders.entity

### PR DESCRIPTION
**Pull request by AuroraHuang22**

### CODE REVIEW

File: components/StakeholderInfo.vue [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <div>
+    <div class="font-normal text-[16px] leading-[22px]">
+      <div v-if="address">
+        {{ `cosmos${address.slice(11)}` }}
+      </div>
+      <div v-else-if="isAddress">
+        {{ isAddress }}
+      </div>
```
Comment (nwingt) [inline components/StakeholderInfo.vue:9]: What is the for?

File: utils/cosmos/iscn/sign.ts [hunk 2]
```diff
@@ -89,26 +101,33 @@ function formatISCNPayload(payload: ISCNSignPayload, version = 1) {
...
+  const stakeholders = []
   if (authorNames.length) {
     for (let i = 0; i < authorNames.length; i += 1) {
-      const authorName = authorNames[i];
-      const authorUrl = authorUrls[i];
-      const isNonEmpty = authorUrl || authorName;
+      const authorName = authorNames[i]
+      const authorUrl = authorUrls[i]
+      const authorId = authorWallets[i]
+      const isNonEmpty = authorUrl || authorName
```
Comment (williamchong) [inline utils/cosmos/iscn/sign.ts:115]: Any one of the author field exists then it should be isNonEmpty

File: utils/cosmos/iscn/sign.ts [hunk 3]
```diff
@@ -89,26 +101,33 @@ function formatISCNPayload(payload: ISCNSignPayload, version = 1) {
...
-            name: authorName,
-          },
-          rewardProportion: 1,
-          contributionType: 'http://schema.org/author',
-        }), 'utf8'))
+        stakeholders.push(
+          Buffer.from(
+            JSON.stringify({
+              entity: {
+                '@id': `did:cosmos:${authorId.slice(6)}` || undefined,
```
Comment (williamchong) [inline utils/cosmos/iscn/sign.ts:121]: id should not be left as empty
if wallet not exists, use url
if wallet nor url exists, use name as id

**Review by williamchong — CHANGES_REQUESTED** [review]

**Review by williamchong — APPROVED** [review]

File: components/StakeholderInfo.vue [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  @Prop(String) readonly id: string | undefined
+  @Prop(String) readonly address: string | undefined
+  @Prop(String) readonly name: string | undefined
+  @Prop(String) readonly url: string | undefined
+
+  get isAddress() {
+    if (this.id && this.id.includes('cosmos'))
+      return `cosmos${this.id.slice(11)}`
+    return false
+  }
```
Comment (nwingt) [inline components/StakeholderInfo.vue:39]: Prefer not use isXXXX for non boolean
```suggestion
  get walletAddress() {
    if (this.id && this.id.includes('cosmos')) {
      return `cosmos${this.id.slice(11)}`
    }
    return ''
  }
```

**Review by nwingt — CHANGES_REQUESTED** [review]

**Pull request merged**